### ISSUE TITLE
Matter Smart Home

### ISSUE THREAD

**Issue by mgernhard** [ioBroker/AdapterRequests#699]
There will be coming "matter" made by a cooperation of amazon, apple, google, Zigbee and many more.
https://matter-smarthome.de/en/matter-faq-en/
https://en.wikipedia.org/wiki/Matter_(standard)

https://buildwithmatter.com/

This is an upcoming project: amazon and google are opening the doors in a short time and will hand out some documentation. There aren't devices yet which support matter but there are coming new devices and updates for old devices.  

This project will be based on the WiFi-protocoll. But philips hue e.g. will be able to support matter with a bridge as well. Unfortunally I don't have more information at the moment, which can't be read at the urls.

Thanks. I love this community with all these adapters.

**Comment by UncleSamSwiss** [comment]
This will for sure be integrated in ioBroker.

The current state of the standard (as a written document) is not yet available, but the **entire** implementation (including simulators and code running on Raspberry Pi and even more embedded hardware from Nordic and others) can be found here: https://github.com/project-chip/connectedhomeip

Also, we had a rather long discussion in the forum: https://forum.iobroker.net/topic/40215/thread-co-project-connected-home-over-ip

**Comment by plotpoint** [comment]
What is the current state of matter integration? Matter 1.0 has been released.

**Comment by Apollon77** [comment]
In progress .. We currently start to support with a node-js library for matter integration ... so will still take time, but we work on it

**Comment by Apollon77** [comment]
PS: Anyone that also want to contribute to the library or adapter or sponsor Matter capable devices please write here or contact me. it will be a huge amount of work and effort for this!

**Comment by Newan** [comment]
Would help to the extent of my possibilities to develop or test the adapter

**Comment by Apollon77** [comment]
For sure I will post here once something is available

**Comment by einarmiger0815** [comment]
> PS: Anyone that also want to contribute to the library or adapter or sponsor Matter capable devices please write here or contact me. it will be a huge amount of work and effort for this!

An das Spnsoring welcher Hardware hast du gedacht?

**Comment by Apollon77** [comment]
> An das Spnsoring welcher Hardware hast du gedacht?

Momentan gibts noch nicht viel :-) Wenn sich das ändert kann man denke ich die Frage besser beantworten. Den neuen Ikea Hub hab ich schon gekauft. Alexa gräte und nen Hue Hub (der wohl auch ein FW Update kriegen soll) hab ich auch. Einen Apple Homepod mini zum testen hab ich auch bereits bekommen. Damit ist"hub" erstmal abgedeckt.

Was fehlt - wenn es die dann mal gibt sind "echte" Endgeräte - bevorzugt erstmal WLAN, später ggf Thread/BLE. Ich würde hoffen das zB die Eve Energy sowas wird, aber die hat kein WLAN.
Also ist die Frage wann es was gibt (hier steht fast nur angekündigt - https://matter-smarthome.de/diese-geraete-produkte-sind-kompatibel-mit-matter/ )

**Comment by skoethe76** [comment]
Matter software-code is open source and available here:
[https://github.com/project-chip/connectedhomeip](https://github.com/project-chip/connectedhomeip)

**Comment by skoethe76** [comment]
Vielleicht ein Ansatz:
https://github.com/TheMrBooyah/hassio-dirigera-client

**Comment by Apollon77** [comment]
@skoethe76 If we would have a Dirigra adapter yes, but thats not matter .. so irrelevant for this case

**Comment by SirVival71** [comment]
It would be very interesting to have a possibility to let a raspberry pi act as a thread device.
I have an AppleTV and some EVE device -> The network exists. Now it would be very interesting to add some special devices. I would like to have a pi reading optical interface (IR) of the house entrance energy counter and route these information as a Thread device through the Thread network.
If allready found a border router implementation for the PI but no interface to act as a Thread device.

**Comment by Apollon77** [comment]
Ok, Why Thread? WOuld not be "Ethernet" as transprt medium completly sufficient?

The plan is to have "soon" a Matter adapter that offers devices via Ethernet/Wifi as first ... Pairing of other Matter devices via Thread is a later step ... 

Thread is usually encapsulated by a Boprder Router that will translate Ethernet <--> Thread, so ideally we only need Ethernet/IPv6 and the rest is handled by such a Border Router. I also plan to experiemtn around with that somewhen

**Comment by SirVival71** [comment]
Yes, Ethernet is the best. But in the case of my house energy counter. There is no Ethernet in the room and it would be good if I have not to install it there. The next Ethernet is 3 rooms away from there.
I do no home automation by WLAN because I've switched this off at night and if nobody is at home.

But I have: A Thread network through the house, a zwave and only for lights a ZigBee by Hue.
For future purposes Thread devices are preferred because of the compatibility (where no Ethernet access is).
But the raspberry or esp32 has to act a Thread device to have it as a compatible device in Apple Home. That's what I'm interested in.
But take the time you need. And follow your plans. I will have a look on it when there is an implementation.

**Comment by msschl** [comment]
Is there any process on this? Is there any thing we could do to help to advance this?

**Comment by mcm1957** [comment]
@Apollon77 
You can answer this best

**Comment by Apollon77** [comment]
In fact right now we do basic implementation of mtter protocol in JavaScript ... driven in https://github.com/project-chip/matter.js

Short ago we reached a milestone that we also can now start on ioBroker side to utilize this work from the last 9 months basically ... I would assume we will have a first basic adapter within next 1-2 months.

If you can develop amnd are interested ... join matter.js :-)

**Comment by layeraccess** [comment]
A Firmware Update for the Nuki door Opener 4 was released with thread support.

**Comment by mcm1957** [comment]
@Apollon77

**Comment by Apollon77** [comment]
Yes I alsoready have a Nuki device sponsored by someone here and could pair it via thread ... sooo ... will work :-)
